pkg/pgmodel/migrations/sql/idempotent/base.sql

```diff
@@ -886,6 +894,118 @@ LANGUAGE SQL VOLATILE;
 
+CREATE OR REPLACE FUNCTION SCHEMA_CATALOG.get_metric_compression_setting(metric_name TEXT)
+RETURNS BOOLEAN
+AS $$
+DECLARE
+    can_compress boolean;
+    result boolean;
+    metric_table_name text;
+BEGIN
+    SELECT exists(select * from pg_proc where proname = 'compress_chunk')
+    INTO STRICT can_compress;
+
+    IF NOT can_compress THEN
+        RETURN FALSE;
+    END IF;
+
+    SELECT table_name 
+    INTO STRICT metric_table_name
+    FROM SCHEMA_CATALOG.get_or_create_metric_table_name(metric_name);
+
+    SELECT EXISTS ( 
+        SELECT FROM _timescaledb_config.bgw_policy_compress_chunks p
+        INNER JOIN _timescaledb_catalog.hypertable h 
+        ON (h.id = p.hypertable_id) 
+        WHERE h.schema_name = 'SCHEMA_DATA' 
+        AND h.table_name = metric_table_name)
+    INTO result;
+    RETURN result;
+END
+$$
+LANGUAGE PLPGSQL STABLE PARALLEL SAFE;
+GRANT EXECUTE ON FUNCTION SCHEMA_CATALOG.get_metric_compression_setting(TEXT) TO prom_reader;
+
+CREATE OR REPLACE FUNCTION SCHEMA_PROM.set_default_compression_setting(compression_setting BOOLEAN)
+RETURNS BOOLEAN
+AS $$
+DECLARE
+    can_compress BOOLEAN;
+BEGIN
+    SELECT exists(select * from pg_proc where proname = 'compress_chunk')
+    INTO STRICT can_compress;
+
+    IF NOT can_compress AND compression_setting THEN
+        RAISE NOTICE 'Cannot enable metrics compression, feature not found';
+        RETURN FALSE;
+    END IF;
+    
+    INSERT INTO SCHEMA_CATALOG.default(key, value) VALUES('metric_compression', compression_setting::text)
+    ON CONFLICT (key) DO UPDATE SET value = EXCLUDED.value;
+    RETURN true;
+END
+$$
+LANGUAGE PLPGSQL VOLATILE;
+COMMENT ON FUNCTION SCHEMA_PROM.set_default_compression_setting(BOOLEAN)
+IS 'set the compression setting for any metrics (existing and new) without an explicit override';
+
+CREATE OR REPLACE FUNCTION SCHEMA_PROM.set_metric_compression_setting(metric_name TEXT, new_compression_setting BOOLEAN)
+RETURNS BOOLEAN
+AS $func$
+DECLARE
+    can_compress boolean;
+    metric_table_name text;
+BEGIN
+    SELECT exists(select * from pg_proc where proname = 'compress_chunk')
+    INTO STRICT can_compress;
+
+    --if already set to desired value, nothing to do
+    IF SCHEMA_CATALOG.get_metric_compression_setting(metric_name) = new_compression_setting THEN
+        RETURN TRUE;
+    END IF;
+
+    --if compression is missing, cannot enable it
+    IF NOT can_compress AND new_compression_setting THEN
+        RAISE NOTICE 'Cannot enable metrics compression, feature not found';
+        RETURN FALSE;
+    END IF;
+
+    --use get_or_create_metric_table_name because we want to be able to set /before/ any data is ingested
+    --needs to run before update so row exists before update.
+    SELECT table_name 
+    INTO STRICT metric_table_name
+    FROM SCHEMA_CATALOG.get_or_create_metric_table_name(set_metric_compression_setting.metric_name);
+
+    IF new_compression_setting THEN
+        EXECUTE format($$
```

Comment:
I think we need to create a function for this so the code to enable compression is not duplicated both here and on line 88